ApplicationLibCode/ProjectDataModel/StimPlanModel/RimStimPlanModel.cpp

```diff
@@ -795,8 +795,17 @@ void RimStimPlanModel::updatePerforationInterval()
             m_thicknessDirectionWellPath->perforationIntervalCollection()->appendPerforation( m_perforationInterval );
         }
 
+        double halfPerforationLength = m_perforationLength() * 0.5;
+        if ( m_fractureOrientation == FractureOrientation::ALONG_WELL_PATH )
+        {
+            // Adjust perforation interval for longitudinal fractures to correct TVD depth
+            cvf::Vec3d wellPathTangent = wellPath()->wellPathGeometry()->tangentAlongWellPath( m_MD() );
```

Comment:
Is it safe to assume these pointers are never nullptr?

Reply:
Yes, it should be. But I have added asserts to indicate that I have thought about it! :smile: